documentation/SupportingDocuments/Population Density API - Algorithm Initial Proposal.md

```diff
@@ -0,0 +1,35 @@
+# Population Density Data API
+## Algorithm initial proposal
+
+This document serves as a reference model for the population density calculation algorithm. It provides an illustrative high-level overview of the steps proposed to determine population density. Designed for developers, this guide is created to enhance understanding and implementation of the algorithm. While it aims to enhance understanding and implementation, it is the responsibility of the developer to create their own algorithm to calculate population density.
+
+### Inputs for the algorithm:
+* UEs connection records data is received pseudonymized or hashed.
+* Requested Space is received in the API customers’ request.
+### Process is as follows:
+<ol>
+<li><strong>Cleanup</strong>: Considering one year of historical information, repeated records will be deleted and IOT (non-personal devices) lines records are filtered due to they do not contribute to the process of estimating future population density.<br>
+<strong>Output 1 </strong>&rarr; Cleaned list of UEs connection records</li><br>
+<li><strong>Attribution:</strong> A cell is associated with each user in each time interval.
+<ol><li>For mobile lines with persistent records, the last cell in which each user has been connected in each interval (considering intervals of 15 minutes) will be associated.</li>
+<li>It is common to have intervals for which a user does not show activity, for
+them it will be assumed that the user remains in the same cell in which he was last seen active.</li></ol>
+<strong>Output 2</strong> &rarr; Valid UEs per cell in each time interval</li><br><br>
+<li><strong>Counting and Aggregation:</strong> A count of users is performed per cell and interval.
+On this count, a statistical analysis is performed with the elimination of outliers.
+Records associated with cells with less than K users in each time interval (k –
```

Comment:
This looks like a too early step to apply k-anonymity. One will need to do this again later (after data will be split per grid) anyway.
Suggestion: apply k-anonymity to the final result only.

Reply:
to be further reviewed offline first

Reply:
This has been reviewed with our legal team, and the way to do it right was at this point because if you do it in a later step you are introducing noise to the estimation. The reason why is, as the users are distributed homogeneously (among the grids associated with the cell) the k-anonymity will be biased as the distribution is not 100% a faithful representation. So, from a legal point of view and to not introduce this noise, the right step to do so would be step 3 as this is the last time the data reflects 100% the reality.

Reply:
This approach can cause "over-anonymization" on the API level, but legal team opinion is a strong argument.

Then we have to change k-anonymity threshold from "number of devices" to "density".
Like "Records associated with cells with less than K devices per square km in each time interval".
Current text can be read as k-anonymity is "X users per cell". 
But 
1) cells can have different sizes.
2) requested area can cover only half of the cell. So "100 devices in the cell" will be scaled down to 50 devices on the next step.

But explicitly describe k-anonymity threshold as devices/km2 addresses both problems.

Reply:
As mentioned, the legal recommendation is to keep on applying the k-anonimity over the users in a cell in step 3 (Counting and Aggregation). 

Implementing this at a later stage could lead to two potential problems: (1) Small towns with low populations might be excluded from the estimation if large cells cover these entire towns, where the small population is dispersed across multiple geohashes, and (2) There could be inaccuracies in the estimation. If the population density is lower than the k-anonymity threshold, applying k-anonymity to the geohashes and using the threshold as the population value could result in that the estimated population would be greater than the real population of the country. Conversely, if we choose to return a value of 0, the overall population density of the entire country would appear lower than it actually is. Therefore, the proposal of Implementing the k-anonimity at a later stage is not being considered.

On the other hand, we find it reasonable to implement k-anonymity based on km2 and time intervals. This approach will be incorporated into the Algorithm Proposal for assessment by the respective legal departments and in accordance with the local regulations of each country where it will be applied.